codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java

```diff
@@ -544,6 +544,11 @@ public static void toHttp2Headers(HttpHeaders inHeaders, Http2Headers out) {
                             do {
                                 out.add(COOKIE, value.subSequence(start, index, false));
                                 // skip 2 characters "; " (see https://tools.ietf.org/html/rfc6265#section-4.2.1)
+                                if (index + 1 >= value.length() || value.charAt(index + 1) != ' ') {
+                                    throw new IllegalArgumentException(
+                                            "cookie value has a semicolon not followed by a space, counter to spec: "
+                                                    + value);
+                                }
                                 start = index + 2;
```

Comment:
I think this method was not expected to throw exceptions at all so starting to do so now doesn't seem quite right. As you've noted the `SP` token isn't universally adhered to so it feels like the right thing to do is be lenient on what we accept and not worry require the space.
If we really want to adhere to the spec then it seems that if we can't pull apart the cookie header into multiple individual headers we should just not pull it apart at all and add the whole thing as-is. The only reason we pull it apart is for a HPACK optimization.

Reply:
I agree with @bryce-anderson

Reply:
the method already throws an exception before this patch for the input `one=foo;`

Reply:
That's true. I'd personally prefer if we just added the header as it was in that case as well since we accepted the invalid cookie already in the `HttpHeaders`. More specific to this PR, do you think it's better to throw an exception or not require the `SP`? These conversion are often done async so it can be tough to track down the problem if we throw (as I'm sure you're aware since you dug this up 😅).

Reply:
i think by robustness principle, we should throw an error. it is most likely that http2 is the network side and http1 is the app side. since this is http1->http2 conversion, that means the header field was likely produced by the app and thus we should be strict in how we pass it on to the client.

i dont think it is difficult to track down with this error message, in my case it was obvious where to look once i saw the error.

Maybe it is worthwhile after all to fall back to not separating the string if it is malformed, even in the existing error case. But looking at the use sites, it seems that they are already doing some other checks that can yield exceptions, such as parsing of request.uri() and of course header validation.

Reply:
My take is that I'd prefer not to throw a new exception since we already accepted the header into the `HttpHeaders inHeaders`. I'd personally lean toward just not splitting if we run into any problems trying to do so but I admit thats an opinion.

All that said, if we decide to throw an exception this implementation looks correct to me.